### PR TITLE
Add a pacman action

### PR DESCRIPTION
**Pull request by eds-collabora**

### CODE REVIEW

**Comment by eds-collabora** [comment]
Summary of comments on this code to date:

- [ ] [Drop "--color never" from the pacman invocation](https://github.com/santoshmahto7/debos/commit/994a093e6e56ecbd875efb570e44ecb5fefd5e39#r53536394)
- [ ] [Drop timeout patch](https://github.com/santoshmahto7/debos/commit/25634f477acf727ab3375086c866bcc499fb0743#commitcomment-53536450)

File: actions/pacman_action.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+type PacmanAction struct {
+	debos.BaseAction `yaml:",inline"`
+	Packages         []string
+}
+
+func (p *PacmanAction) Run(context *debos.DebosContext) error {
+	p.LogStart()
+
+	pacmanOptions := []string{"pacman", "--color", "never", "--noprogressbar", "--noconfirm", "-Syu"}
+	pacmanOptions = append(pacmanOptions, p.Packages...)
```
Comment (eds-collabora) [inline actions/pacman_action.go:31]: Drop --colour never (original comment [here](https://github.com/santoshmahto7/debos/commit/994a093e6e56ecbd875efb570e44ecb5fefd5e39#r53536394))

File: actions/pacman_action.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ - action: pacman
+   packages:
+     - package1
+     - package2
+
+Mandatory properties:
+
+- packages -- list of packages to install
+*/
+package actions
```
Comment (obbardc) [inline actions/pacman_action.go:16]: wondering if we should have an implementation for `Verify` which checks if pacman (and other used) commands are present on the system; if not we should bail out early?

**Review by evelikov — CHANGES_REQUESTED** [review]

File: actions/pacman_action.go (hunk 2 above)
Comment (eds-collabora) [reply actions/pacman_action.go:16]: I think (from the pacstrap PR) we wanted to make that a separate follow-up. I have the code, but perhaps doesn't belong in this PR?

File: actions/pacman_action.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+func (p *PacmanAction) Run(context *debos.DebosContext) error {
+	p.LogStart()
+
+	pacmanOptions := []string{"pacman", "--noprogressbar", "--noconfirm", "-Syu"}
+	pacmanOptions = append(pacmanOptions, p.Packages...)
+
+	c := debos.NewChrootCommandForContext(*context)
+	err := c.Run("Pacman", pacmanOptions...)
+	if err != nil {
```
Comment (obbardc) [inline actions/pacman_action.go:35]: you could merge this line with the line above to improve readability

**Comment by evelikov** [comment]
The original comment suggests dropping the `color` and `progressbar` cmdline options, plus some cosmetics. Feel free to ignore the cosmetics, but please drop the `progressbar` option ... or add an inline comment why we need it. With that, the MR is:
```
Reviewed-by: Emil Velikov <emil.l.velikov@gmail.com>
```

**Comment by eds-collabora** [comment]
> The original comment suggests dropping the `color` and `progressbar` cmdline options, plus some cosmetics. Feel free to ignore the cosmetics, but please drop the `progressbar` option ... or add an inline comment why we need it.

I just didn't read the original comment carefully enough, sorry. Fixed.

**Comment by obbardc** [comment]
closing in favour of !377

**Pull request closed without merging**